modules/ROOT/pages/syntax-quick-reference.adoc

```diff
@@ -0,0 +1,1290 @@
+----
+
+.View result of <<ex-check>>
+[%collapsible.result]
+====
+include::lists:example$checklist.adoc[tag=check]
+====
+
+.Description list
+[#ex-dlist]
+----
+include::lists:example$description.adoc[tag=qr-base]
+----
+
+.View result of <<ex-dlist>>
+[%collapsible.result]
+====
+include::lists:example$description.adoc[tag=qr-base]
+====
+
+.Question and answer list
+[#ex-qa]
+----
+include::lists:example$description.adoc[tag=qa]
+----
+
+.View result of <<ex-qa>>
+[%collapsible.result]
+====
+include::lists:example$description.adoc[tag=qa]
+====
+
+.Mixed
+[#ex-mixed]
+----
+include::lists:example$description.adoc[tag=3-mix]
+----
+
+.View result of <<ex-mixed>>
+[%collapsible.result]
+====
+include::lists:example$description.adoc[tag=3-mix]
+====
+
+TIP: Lists can be indented.
+Leading whitespace is not significant.
+
+.Complex content in outline lists
+[#ex-complex]
+----
+include::lists:example$complex.adoc[tag=b-complex]
+----
+
+.View result of <<ex-complex>>
+[%collapsible.result]
+====
+include::lists:example$complex.adoc[tag=b-complex]
+====
+
+== Links
+
+.Autolinks, URL macro, and mailto macro
+[#ex-urls]
+----
+include::macros:example$url.adoc[tag=b-base]
+
+include::macros:example$url.adoc[tag=b-scheme]
+----
+
+.View result of <<ex-urls>>
+[%collapsible.result]
+====
+include::macros:example$url.adoc[tag=b-base]
+
+include::macros:example$url.adoc[tag=b-scheme]
+====
+
+.URL macros with attributes
+[#ex-linkattrs]
+----
+include::macros:example$url.adoc[tag=b-linkattrs]
+----
+
+.View result of <<ex-linkattrs>>
+[%collapsible.result]
+====
+include::macros:example$url.adoc[tag=b-linkattrs]
+====
+
+CAUTION: If the linked text contains a comma and the text is followed by one or more named attributes, you must enclose the text in double quotes.
+Otherwise, the text will be cut off at the comma (and the remaining text will get pulled into the attribute parsing).
+
+.URLs with spaces and special characters
+----
+include::macros:example$url.adoc[tag=b-spaces]
+----
+
+.Link to relative page
+----
+link:index.html[Docs]
+----
+
+.Link using a Windows path
+----
+include::macros:example$url.adoc[tag=b-windows]
+----
+
+.Inline anchors
+----
+include::attributes:example$id.adoc[tag=anchor]
+----
+
+.Cross references
+[#ex-xrefs]
+----
+include::macros:example$xref.adoc[tag=b-base]
+----
+
+.View result of <<ex-xrefs>>
+[%collapsible.result]
+====
+include::macros:example$xref.adoc[tag=b-base]
+====
+
+.Inter-document cross references
+----
+include::macros:example$xref.adoc[tag=b-inter]
+----
+
+== Images
+
+Images are resolved relative to the value of the xref:macros:images-directory.adoc[imagesdir] document attribute, which is empty by default.
+Use the `imagesdir` attribute to avoid hard coding the common path to your images in every image macro.
+The `imagesdir` attribute can be an absolute path, relative path, or base URL.
+When the image target is a URL or absolute path, the imagesdir prefix is _not_ prepended.
```

Comment:
We might want to use monospaced consistently on `imagesdir` (unless I'm missing a rule of thumb)?

Reply:
This is something the style guide is going to need to address. Sarah hates the over-monospacification of terms. Her rule is usually to use monospaced when first introducing it, then dropping it for the remainder of the text because the reader understands that's the subject. It's not done that way here, though it presents a good opportunity to exercise the rule.

Reply:
> Sarah hates the over-monospacification of terms.

Umm, who is this Sarah you're talking about? :grin: I've evolved on monospacing. First, because it helps differentiate when we're talking about a concept that happens to share the same name as an attribute, option, method, etc. Now, I try to always mark it up as monospace when its the attribute, option, method, etc. when it's in regular text. My personal exceptions to this are when it's in a title (like a section title or listing title) or in linked text, because I think the double formatting, usually with a larger font and additional colors, makes it stand out too much and causes visual distraction. For instance:

```
== imagesdir attribute // I wouldn't monospace it here

xref:macros:images-directory.adoc[imagesdir] // or here

.Set the imagesdir attribute // or here
----
an example
----
```

However, I am open to alternative guidelines. While I find it visually distracting to add monospace formatting on top of bold, bigger font, blue underlined text, etc., and therefore don't use it in such circumstances, I also understand that this may not be the case for most people.

Reply:
Thanks for clarifying your position, @graphitefriction. I'm eating my words now. This seems like a great candidate for the documentation style guide. I'll add it there.